packages/snaps-execution-environments/src/common/validation.test.ts

```diff
@@ -78,3 +79,89 @@ describe('assertIsOnTransactionRequestArguments', () => {
     },
   );
 });
+
+describe('sanitizeJsonStructure', () => {
+  it('should return sanitized JSON', () => {
+    // Make sure that getters cannot have side effect
+    const testSubject = { a: {}, b: {} };
+    let counter = 0;
+    Object.defineProperty(testSubject, 'jailbreak', {
+      enumerable: true,
+      get() {
+        counter += 1;
+        return counter;
+      },
+      set(value) {
+        return (counter = value);
+      },
+    });
+    const result = sanitizeJsonStructure(testSubject) as { jailbreak: number };
+
+    // Remember the counter
+    const finalCount = result.jailbreak;
+    expect(typeof result.jailbreak).not.toBe('function');
```

Comment:
why would it ever be a function? even if you get the original one, it's still not gonna be a function.

Reply:
You shouldn't need more assertions than this:
```js
  // check that the counter is not increasing
  expect(result.jailbreak).toEqual(result.jailbreak);

  // check that it's a value not a getter explicitly
  const desc = Object.getOwnPropertyDescriptor(result, 'jailbreak');
  expect(desc). to have a value not get/set pair
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptor

Reply:
Refactored the test according to @naugtur suggestion.